articles/tech-resources/deep-links-self-serve.md

```diff
@@ -0,0 +1,189 @@
+
+The setup steps are as follows:
+
+1. Configure a CDN to host your custom tracking domain.  One example can be found [here](https://www.sparkpost.com/docs/tech-resources/enabling-https-engagement-tracking-on-sparkpost/).
+**Note:** The CDN needs to support file hosting such as AWS CloudFront.
+1. [Configure a custom tracking domain on your SparkPost account.](https://app.sparkpost.com/account/tracking-domains)
+1. [Use a "custom link sub-path" in your links.](https://developers.sparkpost.com/api/substitutions-reference.html#header-custom-link-sub-paths)
+1. Publish the `apple-app-site-associate` or `assetlinks.json` file for your app using both the tracking domain and link sub-path.  These should be hosted on your CDN under  **\<your-tracking-domain>\\.well-known\\**. Your apple-app-site-association file might include the following:
+    ```javascript
+    {
+      "applinks": {
+        "apps": [],
+        "details": [
+          {
+            "appID": "<YOUR_APP_ID>",
+            "paths": [
+              "/f/open-in-app/*"
+            ]
+          }
+        ]
+      }
+    }
+    ```
+1. Send your email through SparkPost with click tracking enabled, using your custom tracking domain and link sub-path:
+    ```
+    Your invoice is available <a data-msys-sublink="open-in-app" href="http://www.example.com/invoice/101">here</a>.
+    ```
+1. Update your mobile app to make an HTTP GET request to the SparkPost click tracking service when it handles the incoming click event. There are more details and sample code for this below.
+1. (Optional) Have your app take action based on the original URL which SparkPost returns in its "3xx redirect" HTTP response.
+
+#### Forwarding Clicks From iOS To SparkPost
+
+When an iOS email client recognizes that your recipient has tapped on a universal link, it sends your application delegate an `NSUserActivity`. To allow SparkPost to track this universal link "click" event, you must call the SparkPost click tracking service to tell it about the activity. The click tracker will respond with a "3xx" redirect to the original URL from your email which may be useful to control app behaviour.
+
+Here's a sample `application:continueUserActivity:restorationHandler` method which calls the SparkPost click tracker to register a click event and also retrieves the original URL:
+
+```objc
+- (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserActivity *)userActivity
+ restorationHandler:(void (^)(NSArray *))restorationHandler {
+
+    // Ignore non-universal link activities
+    if (![userActivity.activityType isEqualToString:NSUserActivityTypeBrowsingWeb] || userActivity.webpageURL == nil) {
+        return NO;
+    }
+
+    // Create a task to call the SparkPost click tracker, expecting a 3xx redirect to your original URL.
+    // On completion, the handler will receive an NSURLResponse which knows the original target URL.
```

Comment:
With a CDN you can just do a GET on the original Tracking link, so I removed that substitution but hopefully the objC is still valid.

Reply:
More Info:  
With a CDN, it is no longer required to redirect the link to spgo.io as the tracking domain will successfully forward the click.  Direction for this ticket was to focus on the use of a CDN versus a method of removing CNAME method.